students/Dkuchan/session07/circles.py

```diff
@@ -0,0 +1,53 @@
+#circles.py
+from math import pi
+
+
+class Circle(object):
+
+    def __init__(self, radius):
+        self.radius = float(radius)
+
+    @property
+    def diameter(self):  # Getter has to return a variable
+        return 2 * self.radius
+
+    @diameter.setter    # Setter is mutable based on new data
+    def diameter(self, d):
+        self.radius = d / 2
+
+    @property
+    def area(self):
+        return(pi * self.radius**2)
+
+    @property
+    def circumference(self):
+        return(2 * pi * self.radius)
+
+    @classmethod
+    def from_diameter(cls, diam):
+        return cls(diam / 2)
+
+
+class Sphere(Circle):
+
+    def volume(self):
+        return (4 / 3 * pi * self.radius ** 3)
+
+    def area(self):
+        return (4 * pi * self.radius ** 2)
+
```

Comment:
Since area is a property in the Circle object, and volume is very similar, they should both be properties here too.